File: problems/intermediate/ema_supercomputer/__tests__/test_ema_supercomputer.py
import sys
import os

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from ema_supercomputer import twoPluses

def test_two_pluses():
    grid = [
        "GGGGGG",
        "GBBBGB",
        "GGGGGG",
        "GGBBGB",
        "GGGGGG"
    ]
    result = twoPluses(grid)
    assert result == 5, f"Expected 5, got {result}"
    print("✅ Python test passed!")

if __name__ == "__main__":
    test_two_pluses()

File: problems/intermediate/ema_supercomputer/__tests__/TestEmaSupercomputer.java
package problems.intermediate.ema_supercomputer.__tests__;

import problems.intermediate.ema_supercomputer.EmaSupercomputer;

public class TestEmaSupercomputer {
     public static void main(String[] args) {
        String[] grid = {
            "GGGGGG",
            "GBBBGB",
            "GGGGGG",
            "GGBBGB",
            "GGGGGG"
        };

        int result = EmaSupercomputer.twoPluses(grid);

        if (result != 5) {
            throw new AssertionError("❌ Test failed. Expected 5, got " + result);
        }

        System.out.println("✅ Java test passed!");
    }
}
